ocpmodels/trainers/dist_forces_trainer.py

```diff
@@ -328,23 +328,25 @@ def train(self):
                     split="val", epoch=epoch,
                 )
 
-            self.best_val_mae = min(
-                val_metrics.meters["force_z/mae"].global_avg, self.best_val_mae
-            )
-            if not self.is_debug and distutils.is_master():
-                save_checkpoint(
-                    {
-                        "epoch": epoch + 1,
-                        "state_dict": self.model.state_dict(),
-                        "optimizer": self.optimizer.state_dict(),
-                        "normalizers": {
-                            key: value.state_dict()
-                            for key, value in self.normalizers.items()
-                        },
-                        "config": self.config,
-                    },
-                    self.config["cmd"]["checkpoint_dir"],
+            if iters % eval_every == 0 or eval_every == -1:
+                self.best_val_mae = min(
+                    val_metrics.meters["force_z/mae"].global_avg,
+                    self.best_val_mae,
                 )
+                if not self.is_debug and distutils.is_master():
+                    save_checkpoint(
+                        {
+                            "epoch": epoch + 1,
+                            "state_dict": self.model.state_dict(),
+                            "optimizer": self.optimizer.state_dict(),
+                            "normalizers": {
+                                key: value.state_dict()
+                                for key, value in self.normalizers.items()
+                            },
+                            "config": self.config,
+                        },
+                        self.config["cmd"]["checkpoint_dir"],
+                    )
```

Comment:
- Not sure if I'd missed this earlier, but this checkpoint saving logic looks incorrect. We only want to save checkpoints when accuracy on val improves, i.e. `val_metrics.meters["force_z/mae"].global_avg` is better than `self.best_val_mae`. Right now it'll save checkpoints at every epoch even if val accuracy gets worse.

- And we should probably move this checkpoint saving logic to within the `if iters % eval_every == 0 and eval_every != -1:` and `if eval_every == -1:` blocks on L309 and L316 respectively.